### PR TITLE
ESLint update - Set up with @vercel/style-guide

### PR DESCRIPTION
**Pull request by cdedreuille**
WAYWO

### CODE REVIEW

File: apps/frontpage/scripts/copy-other-sitemaps.ts [hunk 1]
```diff
@@ -34,11 +34,11 @@ const OTHER_SITEMAPS = {
 const DESTINATION = path.join(__dirname, '../public/sitemap');
 const SITEMAP_FILENAME = 'sitemap.xml';
 
-function stripDirname(file: string) {
-  return file.replace(/.*(\/public\/.*)/, '$1');
+function stripDirname(file: string): string {
+  return file.replace(/.*(?<temp1>\/public\/.*)/, '$1');
```
Comment (kylegach) [inline apps/frontpage/scripts/copy-other-sitemaps.ts:38]: What is `<temp1>` here?

Comment (cdedreuille) [reply apps/frontpage/scripts/copy-other-sitemaps.ts:38]: This is what was recommended by ESLint.
> Capture group '(\/public\/.*)' should be converted to a named or non-capturing group.
https://eslint.org/docs/latest/rules/prefer-named-capture-group

File: apps/frontpage/lib/get-page.tsx [hunk 2]
```diff
@@ -72,11 +72,14 @@ export const getPageData = async (
       li: MDX.List,
       pre: MDX.Pre,
       details: () => <details>Hello world</details>,
+      // eslint-disable-next-line react/jsx-pascal-case -- TODO: Not sure why this through an error.
```
Comment (kylegach) [inline apps/frontpage/lib/get-page.tsx:75]: I suspect it wants it to be `<Mdx.Img />`.

File: apps/frontpage/scripts/copy-other-sitemaps.ts (hunk 1 above)
Comment (kylegach) [reply apps/frontpage/scripts/copy-other-sitemaps.ts:38]: Oh, I completely forgot about that feature. Thanks.

**Review by kylegach — APPROVED** [review]
This PR makes me very happy. 🙂

File: apps/frontpage/lib/get-page.tsx (hunk 2 above)
Comment (cdedreuille) [reply apps/frontpage/lib/get-page.tsx:75]: But that's what I did ... But the fact that I'm passing props makes it a problem.

Comment (cdedreuille) [reply apps/frontpage/lib/get-page.tsx:75]: I really don't get it.

File: packages/eslint-config/next.js [hunk 3]
```diff
@@ -1,38 +1,61 @@
...
+    // TODO: Remove this override once the issues are resolved
+    '@typescript-eslint/no-unnecessary-condition': 'off',
+    // TODO: Remove this override once the issues are resolved
+    '@typescript-eslint/explicit-function-return-type': 'off',
+    // TODO: Remove this override once the issues are resolved
+    '@typescript-eslint/no-empty-function': 'off',
+    // TODO: Remove this override once the issues are resolved
+    '@typescript-eslint/no-non-null-assertion': 'off',
+    // TODO: Remove this override once the issues are resolved
+    'react/function-component-definition': 'off',
```
Comment (kylegach) [inline packages/eslint-config/next.js:59]: Reminder to myself to revisit these.

Comment (cdedreuille) [reply packages/eslint-config/next.js:59]: Yes it was just too much to do everything at once and those could be solved one by one in separate PRs.

**Pull request merged**